### PR TITLE
[INLONG-3231][DataProxy] Change the default topic to an optional configuration

### LINKED ISSUE #3231: [Feature][DataProxy] Change the default topic to an optional configuration
### Description

Change the default topic to an optional configuration. If there is no default topic, the corresponding topic cannot be found and discarded directly

### Use case

Change the default topic to an optional configuration. If there is no default topic, the corresponding topic cannot be found and discarded directly

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
Change the default topic to an optional configuration. If there is not default topic, the corresponding topic cannot be found and discarded directly

Fixes #3231

### CODE REVIEW

File: inlong-dataproxy/conf/dataproxy-pulsar.conf [hunk 1]
```diff
@@ -42,12 +42,12 @@ agent1.sources.tcp-source.msg-factory-name = org.apache.inlong.dataproxy.source.
 agent1.sources.tcp-source.host = 0.0.0.0
 agent1.sources.tcp-source.port = 46801
 agent1.sources.tcp-source.max-msg-length = 524288
-agent1.sources.tcp-source.topic = test1
+#agent1.sources.tcp-source.topic = test1
```
Comment (healchow) [inline inlong-dataproxy/conf/dataproxy-pulsar.conf:45]: Why not remove this line?

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/BaseSource.java [hunk 2]
```diff
@@ -240,12 +240,12 @@ public void configure(Context context) {
...
 
-    topic = context.getString(ConfigConstants.TOPIC);
+    topic = context.getString(ConfigConstants.TOPIC, "");
     attr = context.getString(ConfigConstants.ATTR);
-    Configurables.ensureRequiredNonNull(context, ConfigConstants.TOPIC, ConfigConstants.ATTR);
+    Configurables.ensureRequiredNonNull(context, ConfigConstants.ATTR);
 
     topic = topic.trim();
-    Preconditions.checkArgument(!topic.isEmpty(), "topic is empty");
+    //Preconditions.checkArgument(!topic.isEmpty(), "topic is empty");
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/BaseSource.java:248]: If not needed, pls remove unused codes.

File: inlong-dataproxy/conf/dataproxy-pulsar.conf (hunk 1 above)
Comment (baomingyu) [reply inlong-dataproxy/conf/dataproxy-pulsar.conf:45]: it is optional config,  and this can be as example.

Comment (dockerzhang) [inline inlong-dataproxy/conf/dataproxy-pulsar.conf:45]:
```suggestion
# agent1.sources.tcp-source.topic = public/default/dataproxy-topic
```

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**